timemachine/fe/free_energy.py

```diff
@@ -44,90 +145,113 @@ def _get_system_params_and_potentials(ff_params, topology):
 
         return final_params, final_potentials
 
-    def prepare_host_edge(self, ff_params, host_system, host_coords):
+
+# this class is serializable.
+class AbsoluteFreeEnergy(BaseFreeEnergy):
+    def __init__(self, mol, top):
         """
-        Prepares the host-edge system
+        Compute the absolute free energy of a molecule via 4D decoupling.
+
+        Parameters
+        ----------
+        mol: rdkit mol
+            Ligand to be decoupled
+
+        top: Topology
+            topology.Topology to use
+
+        """
+        self.mol = mol
+        self.top = top
+
+    def prepare_host_edge(self, ff_params, host_system):
+        """
+        Prepares the host-guest system
 
         Parameters
         ----------
         ff_params: tuple of np.array
             Exactly equal to bond_params, angle_params, proper_params, improper_params, charge_params, lj_params
 
         host_system: openmm.System
-            openmm System object to be deserialized
-
-        host_coords: np.array
-            Nx3 array of atomic coordinates
+            openmm System object to be deserialized.
 
         Returns
         -------
-        4 tuple
-            unbound_potentials, system_params, combined_masses, combined_coords
+        3-tuple
+            unbound_potentials, system_params, combined_masses
 
         """
-        raise NotImplementedError("Not implemented")
+        ligand_masses = get_mol_masses(self.mol)
+
+        host_bps, host_masses = openmm_deserializer.deserialize_system(host_system, cutoff=1.2)
+        hgt = topology.HostGuestTopology(host_bps, self.top)
 
+        final_params, final_potentials = self._get_system_params_and_potentials(ff_params, hgt)
+        combined_masses = self._combine(ligand_masses, host_masses)
+        return final_potentials, final_params, combined_masses
 
-# this class is serializable.
-class AbsoluteFreeEnergy(BaseFreeEnergy):
-    def __init__(self, mol, ff):
+    def prepare_vacuum_edge(self, ff_params):
         """
-        Compute the absolute free energy of a molecule via 4D decoupling.
+        Prepares the vacuum system
 
         Parameters
         ----------
-        mol: rdkit mol
-            Ligand to be decoupled
+        ff_params: tuple of np.array
+            Exactly equal to bond_params, angle_params, proper_params, improper_params, charge_params, lj_params
```

Comment:
nit: 'Exactly equal' seems like the wrong wording. Perhaps 'Expected parameter ordering is as follows:'?